pb/errors.proto

```diff
@@ -0,0 +1,16 @@
+syntax = "proto3";
+package pb;
+
+// Errors are coded in Rossby to easily identify what has gone wrong
+// TODO: create error code enumerations for quick testing of error types
+message Error {
+    uint32 code = 1;                // the unique error code specific to Rossby
+    string message = 2;             // the error message to display to clients
+}
+
+// Multi-errors store more than one error but can be treated like a regular error
+// Normally we'd implement this as a type []*Error but this format allows it to be
+// easily serialized in RPC requests and responses.
+message MultiError {
+    repeated Error errors = 1;      // the errors in the multi-error
+}
```

Comment:
As I mentioned in the standup, multi errors are the primary usage because there are a lot of contexts where multiple things happen, e.g. sending multiple messages at a time or parsing multiple contacts.

Reply:
This is a neat idea; is there a corresponding design pattern for looping through the resulting errors to diagnose/handle?

Reply:
I would expect that you could loop through each of the errors to handle them, but we will have to put in more thought about how to associate the errors with the things they are referring to. For example, if you send 4 messages but only the second and third fail - in the app you'd want to put a "could not send, try again" message under the message itself, but you'd have to figure out which messages succeeded and failed. Any thoughts you have on that would be welcome!

Reply:
Gotcha &mdash; in that case perhaps the best way to deal with this is for now to keep the logging verbose and watch how the multierrors come up in practice since I'm guessing some we'll be ok with essentially ignoring, while others might prompt the user to try the action again or let them know they don't have the correct permissions to do the action. Hard to know without playing around a bit first